db/migrate/20210626161444_change_fulfillment_date_of_check_to_date.rb

```diff
@@ -0,0 +1,9 @@
+class ChangeFulfillmentDateOfCheckToDate < ActiveRecord::Migration[6.1]
+  def up
+    change_column :fulfillments, :date_of_check, 'date USING date_of_check::date'
```

Comment:
Rails had an under the hood change that was making nils / empty strings error when being cast to dates, and it looks like when I set up fulfillments I set this up as a string instead of a date.

Note that this does preserve dates properly:

Before:
```
dcaf_case_management_development=# select distinct date_of_check from fulfillments; 
       date_of_check       
---------------------------
 
 2021-02-15 01:42:11 -0500
```

After (with an extra row from some experimenting):

```
dcaf_case_management_development=# select distinct procedure_date from fulfillments; 
 procedure_date 
----------------
 
 2021-02-10
 2021-06-30
```

Reply:
that second query isn't for `date_of_check`?

Reply:
this didn't quite work for me out of the box:

```
Sun Jul 11 15:51:05 [r61] ~/repos/dcaf $ make migrate
bundle exec rails db:migrate
warning: parser/current is loading parser/ruby27, which recognizes
warning: 2.7.1-compliant syntax, but you are running 2.7.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
== 20210626161444 ChangeFulfillmentDateOfCheckToDate: migrating ===============
-- change_column(:fulfillments, :date_of_check, "date USING date_of_check::date")
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::InvalidDatetimeFormat: ERROR:  invalid input syntax for type date: ""
/home/lomky/repos/dcaf/db/migrate/20210626161444_change_fulfillment_date_of_check_to_date.rb:3:in `up'

Caused by:
ActiveRecord::StatementInvalid: PG::InvalidDatetimeFormat: ERROR:  invalid input syntax for type date: ""
/home/lomky/repos/dcaf/db/migrate/20210626161444_change_fulfillment_date_of_check_to_date.rb:3:in `up'

Caused by:
PG::InvalidDatetimeFormat: ERROR:  invalid input syntax for type date: ""
/home/lomky/repos/dcaf/db/migrate/20210626161444_change_fulfillment_date_of_check_to_date.rb:3:in `up'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
make: *** [Makefile:62: migrate] Error 1
```

It works when I dropped & created a new DB, but that won't really be an option in prod! I think the issue is non-robust handling of empty string

Reply:
agree with your diagnosis here. I think we can probably coerce a blank string into a null in the context of this migration, I hope? I'll give it a shot tmw.